server/src/main/java/org/opensearch/search/querygroup/QueryGroupTask.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.querygroup;
+
+/**
+ * This interface can be implemented by tasks which will be tracked and monitored using {@link org.opensearch.cluster.metadata.QueryGroup}
+ */
+public interface QueryGroupTask {
```

Comment:
We don;t need this class lets remove this.

Reply:
this is dependant on - https://github.com/opensearch-project/OpenSearch/pull/13897#discussion_r1687030650

Reply:
This is the latest - https://github.com/opensearch-project/OpenSearch/pull/14708#discussion_r1688822067